### DISCUSSION TITLE
Unable to use levenshtein function in deterministic rule for Athena Linker

### DISCUSSION THREAD

**Discussion by roshnaramesh** [moj-analytical-services/splink#1256] [Q&A] [answered]
Hello,
I'm using the Athena linker to perform a link_only comparison between 2 dataframes. I wanted to use the levenshtein() function in the deterministic rule , as mentioned in this example:
`deterministic_rules = [
    "l.first_name = r.first_name and levenshtein(r.dob, l.dob) <= 1",
]
linker.estimate_probability_two_random_records_match(deterministic_rules, recall=0.7)
`
This is my code:
`linker = AthenaLinker(input_table_or_tables=input_tables, boto3_session=boto3_session, output_bucket=output_bucket,
                          output_database=output_database, settings_dict=settings)
`
` deterministic_rules = [
        'levenshtein(r.lastname,l.lastname)<=2'
   ]`
` linker.estimate_probability_two_random_records_match(deterministic_rules, recall=0.7)`
When I run this, it throws this error
`awswrangler.exceptions.QueryFailed: SYNTAX_ERROR: line 29:41: Function levenshtein not registered. You may need to manually clean the data at location '' before retrying. Athena will not delete data in your account.
`

I'm able to achieve this with the DuckDB linker. Can someone help me with this?

**Comment by RossKen** [comment, marked as answer]
Hi @roshnaramesh, this is because the  levenshtein function is named `levenshtein_distance` in Athena's SQL engine. https://trino.io/docs/current/functions/string.html

It should work if you change your deterministic rule code
